app/Console/Commands/video_data.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace App\Console\Commands;
+
+use App\Configs;
+use App\Metadata\Extractor;
+use App\ModelFunctions\PhotoFunctions;
+use App\Photo;
+use Illuminate\Console\Command;
+use Illuminate\Support\Facades\Config;
+
+class video_data extends Command
+{
+	/**
+	 * The name and signature of the console command.
+	 *
+	 * @var string
+	 */
+	protected $signature = 'video_data {count=100 : number of videos to process} {timeout=600 : timeout time requirement}';
+
+	/**
+	 * The console command description.
+	 *
+	 * @var string
+	 */
+	protected $description = 'Generate video thumbnails and metadata if missing';
+
+	/**
+	 * @var PhotoFunctions
+	 */
+	private $photoFunctions;
+
+	/**
+	 * @var Extractor
+	 */
+	private $metadataExtractor;
+
+	/**
+	 * Create a new command instance.
+	 *
+	 * @param PhotoFunctions $photoFunctions
+	 * @return void
+	 */
+	public function __construct(PhotoFunctions $photoFunctions, Extractor $metadataExtractor)
+	{
+		parent::__construct();
+
+		$this->photoFunctions = $photoFunctions;
+		$this->metadataExtractor = $metadataExtractor;
+	}
+
+	/**
+	 * Execute the console command.
+	 *
+	 * @return mixed
+	 */
+	public function handle()
+	{
+		set_time_limit($this->argument('timeout'));
+
+		$this->line(
+			sprintf(
+				'Will attempt to generate up to %s video thumbnails/metadata with a timeout of %d seconds...',
+				$this->argument('count'),
+				$this->argument('timeout')
+			)
+		);
+
+		$photos = Photo::where('type', 'like', 'video/%')
+			->where('width', '=', 0)
+			->take($this->argument('count'))
+			->get()
+		;
+
+		if (count($photos) == 0) {
+			$this->line('No videos require processing');
+			return 0;
+		}
+
+		foreach ($photos as $photo) {
+			$this->line('Processing '.$photo->title.'...');
+			$url = Config::get('defines.dirs.LYCHEE_UPLOADS_BIG').$photo->url;
+
+			if ($photo->thumbUrl != '') {
+				$thumb = Config::get('defines.dirs.LYCHEE_UPLOADS_THUMB').$photo->thumbUrl;
+				if (file_exists($thumb)) {
+					$urlBase = explode('.', $photo->url);
+					$thumbBase = explode('.', $photo->thumbUrl);
+					if ($urlBase !== $thumbBase) {
+						$photo->thumbUrl = $urlBase[0].'.'.$thumbBase[1];
+						rename($thumb, Config::get('defines.dirs.LYCHEE_UPLOADS_THUMB').$photo->thumbUrl);
+						$this->line('Renamed thumb to match the video file');
+					}
+				}
+			}
+
+			if (file_exists($url)) {
+				$info = $this->metadataExtractor->extract($url, $photo->type);
+
+				if ($info['width']) {
+					$this->line('Extracted metadata');
```

Comment:
this line should probably be above the if statement.

Reply:
Actually, the metadata extractor doesn't return a success indicator so we don't know if it succeeded to extract anything until we check the contents of the `info` array. So that's why I wrote it that way. It's still not perfect because I suppose it could be that `width` could not be extracted but the `duration` could... Maybe I'm overthinking it. We could also just drop that message altogether? Let me know...

Reply:
Then shouldn't you test `if ($info['width'] != '')` instead ? :)

Reply:
Good point :smiley: Actually, that field is an int not a string, but it's still not right. I'll fix it.